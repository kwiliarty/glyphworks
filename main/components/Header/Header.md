The application Header.

```jsx
<Header />
```
