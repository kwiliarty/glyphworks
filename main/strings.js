import React from 'react'
import Link from './components/Link'
import Ipa from './components/Ipa'

export const main_title = 'GlyphWorks'
export const ipa_title = '[ɡlɪfwɜ˞ks]'
export const welcome_message = <p>Welcome to <Ipa>[ɡlɪfwɜ˞ks]</Ipa>! Convert <Link href='https://en.wikipedia.org/wiki/X-SAMPA'>X-SAMPA</Link> to <Link href='https://en.wikipedia.org/wiki/International_Phonetic_Alphabet'>IPA</Link> or vice versa.</p>
export const quickstart_examples = <p>For example, if you type &quot;Si: sElz si: SElz&quot; into the X-SAMPA field then you should see &quot;<Ipa>ʃiː sɛlz siː ʃɛlz</Ipa>&quot; in the IPA field, a transcription of <em>she sells sea shells</em>.</p>
