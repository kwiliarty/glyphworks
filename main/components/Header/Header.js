import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

import * as Strings from '../../strings'

import Ipa from '../Ipa'

const StyledHeader = styled.header`
  background-color: ${ props => props.theme.colors[ props.backColor ] || props.backColor };
  color: ${ props => props.theme.colors[ props.foreColor ] || props.foreColor };
  padding: ${ props => props.theme.basePadding };
  h1 {
    margin: 0;
    font-size: 2rem;
  }
  a, a:hover, a:focus, a:active, a:visited {
    color: inherit;
    text-decoration: none;
    &:focus {
      outline: 2px dotted ${ props => props.theme.colors[ props.foreColor ] || props.foreColor };
    }
  }
`

const Header = props => {

  return (
    <StyledHeader backColor={ props.backColor } foreColor={ props.foreColor }>
      <h1>
        <Link to='/'>
          { props.mainTitle } <Ipa>{ props.ipaTitle }</Ipa>
        </Link>
      </h1>
    </StyledHeader>
  )
}

Header.propTypes = {
  /** The background color for the header, the name of a color from the theme */
  backColor: PropTypes.string,
  /** The foreground color for the header, the name of a color from the theme */
  foreColor: PropTypes.string,
  /** The main string to display in the title */
  mainTitle: PropTypes.string,
  /** The IPA string to display in the title */
  ipaTitle: PropTypes.string,
}

Header.defaultProps = {
  backColor: 'darkRed',
  foreColor: 'veryLightBrown',
  ipaTitle: Strings.ipa_title,
  mainTitle: Strings.main_title,
}

export default Header
