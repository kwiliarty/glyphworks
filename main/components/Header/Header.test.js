import React from 'react'
import { render, screen } from 'utils'
import Header from './Header.js'
import * as Strings from '../../strings'

describe( 'The Header component', () => {
  it( 'can display a default title', () => {
    render( <Header /> )
    const banner = screen.getByRole( 'banner' )
    expect( banner ).toHaveTextContent( Strings.main_title )
    expect( banner ).toHaveTextContent( Strings.ipa_title )
    expect( banner ).toMatchSnapshot()
  })

  it( 'can display a custom title', () => {
    render( <Header mainTitle='Custom Title' ipaTitle='some IPA' /> )
    const banner = screen.getByRole( 'banner' )
    expect( banner ).toHaveTextContent( 'Custom Title' )
    expect( banner ).toHaveTextContent( 'some IPA' )
    expect( banner ).toMatchSnapshot()
  })

  it( 'can accept custom colors', () => {
    render(
      <Header foreColor='wheat' backColor='pink' />
    )
    const banner = screen.getByRole( 'banner' )
    expect( banner ).toHaveStyleRule('color', 'wheat')
    expect( banner ).toHaveStyleRule('background-color', 'pink')
    expect( banner ).toMatchSnapshot()
  })
})
