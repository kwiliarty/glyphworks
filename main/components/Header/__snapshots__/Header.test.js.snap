// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component can accept custom colors 1`] = `
.c1 {
  font-family: GentiumPlus,serif;
}

.c0 {
  background-color: pink;
  color: wheat;
  padding: 1rem;
}

.c0 h1 {
  margin: 0;
  font-size: 2rem;
}

.c0 a,
.c0 a:hover,
.c0 a:focus,
.c0 a:active,
.c0 a:visited {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus,
.c0 a:hover:focus,
.c0 a:focus:focus,
.c0 a:active:focus,
.c0 a:visited:focus {
  outline: 2px dotted wheat;
}

<header
  class="c0"
>
  <h1>
    <a
      href="/"
    >
      GlyphWorks
       
      <span
        class="c1"
      >
        [ɡlɪfwɜ˞ks]
      </span>
    </a>
  </h1>
</header>
`;

exports[`The Header component can display a custom title 1`] = `
.c1 {
  font-family: GentiumPlus,serif;
}

.c0 {
  background-color: #621818;
  color: #fff7d4;
  padding: 1rem;
}

.c0 h1 {
  margin: 0;
  font-size: 2rem;
}

.c0 a,
.c0 a:hover,
.c0 a:focus,
.c0 a:active,
.c0 a:visited {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus,
.c0 a:hover:focus,
.c0 a:focus:focus,
.c0 a:active:focus,
.c0 a:visited:focus {
  outline: 2px dotted #fff7d4;
}

<header
  class="c0"
>
  <h1>
    <a
      href="/"
    >
      Custom Title
       
      <span
        class="c1"
      >
        some IPA
      </span>
    </a>
  </h1>
</header>
`;

exports[`The Header component can display a default title 1`] = `
.c1 {
  font-family: GentiumPlus,serif;
}

.c0 {
  background-color: #621818;
  color: #fff7d4;
  padding: 1rem;
}

.c0 h1 {
  margin: 0;
  font-size: 2rem;
}

.c0 a,
.c0 a:hover,
.c0 a:focus,
.c0 a:active,
.c0 a:visited {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus,
.c0 a:hover:focus,
.c0 a:focus:focus,
.c0 a:active:focus,
.c0 a:visited:focus {
  outline: 2px dotted #fff7d4;
}

<header
  class="c0"
>
  <h1>
    <a
      href="/"
    >
      GlyphWorks
       
      <span
        class="c1"
      >
        [ɡlɪfwɜ˞ks]
      </span>
    </a>
  </h1>
</header>
`;
